lib/shelley/src/Cardano/Wallet/Shelley/Pools.hs

```diff
@@ -723,6 +769,34 @@ monitorMetadata tr gp DBLayer{..} = do
         toMicroSecond = (`div` 1000000) . fromEnum
         slotLength = unSlotLength $ getSlotLength gp
         f = unActiveSlotCoefficient (getActiveSlotCoefficient gp)
+    onExit = defaultWorkerAfter tr'
+
+gcDelistedPools
+    :: Tracer IO StakePoolLog
+    -> DBLayer IO
+    -> IO (Maybe [PoolId])  -- ^ delisted pools fetcher
+    -> IO ()
+gcDelistedPools tr DBLayer{..} fetchDelisted = forever $ do
+    lastGC <- atomically readLastMetadataGC
+    currentTime <- getPOSIXTime
+
+    let timeSinceLastGC = currentTime - lastGC
+        sixHours = posixDayLength / 4
+    if timeSinceLastGC > sixHours
+        then do
+            delistedPools <- fmap (fromMaybe []) fetchDelisted
+            atomically $ do
+                putLastMetadataGC currentTime
+                delistPools delistedPools
+        else do
+            -- Sleep for 60 seconds. This is useful in case
+            -- something else is modifying the last sync time
+            -- in the database.
+            let sec_to_milisec = (* 1000000)
+                sleep_time = sec_to_milisec 60
+            traceWith tr $ MsgGCTakeBreak sleep_time
+            threadDelay sleep_time
```

Comment:
```suggestion
    when (timeSinceLastGC > sixHours) $ do
        delistedPools <- fmap (fromMaybe []) fetchDelisted
        atomically $ do
            putLastMetadataGC currentTime
            delistPools delistedPools

    -- Sleep for 60 seconds. This is useful in case
    -- something else is modifying the last sync time
    -- in the database.
    let ms = (* 1000000)
    let sleepTime = ms 60
    traceWith tr $ MsgGCTakeBreak sleepTime
    threadDelay sleepTime
```

Reply:
- You probably want to sleep regardless of whether you did a GC or not.
- Use camelCase for variable names :upside_down_face: